File: reversePolish.js
function evalRPN(tokens) {
    const stack = [];

    for (const token of tokens) {
        if (token === "+" || token === "-" || token === "*" || token === "/") {
            const operand2 = stack.pop();
            const operand1 = stack.pop();
            switch (token) {
                case "+":
                    stack.push(operand1 + operand2);
                    break;
                case "-":
                    stack.push(operand1 - operand2);
                    break;
                case "*":
                    stack.push(operand1 * operand2);
                    break;
                case "/":
                    stack.push(Math.trunc(operand1 / operand2));
                    break;
            }
        } else {
            stack.push(Number(token));
        }
    }

    return stack[0];
}

const tokens = ["2", "1", "+", "3", "*"];
console.log(evalRPN(tokens)); // Output: 9

File: reversePolish.kt
import java.util.Stack

fun evalRPN(tokens: Array<String>): Int {
    val stack = Stack<Int>()

    for (token in tokens) {
        if (token in setOf("+", "-", "*", "/")) {
            val operand2 = stack.pop()
            val operand1 = stack.pop()
            when (token) {
                "+" -> stack.push(operand1 + operand2)
                "-" -> stack.push(operand1 - operand2)
                "*" -> stack.push(operand1 * operand2)
                "/" -> stack.push(operand1 / operand2)
            }
        } else {
            stack.push(token.toInt())
        }
    }

    return stack.pop()
}
